cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemotePluginFinder.java

```diff
@@ -202,16 +198,13 @@ protected Location getArtifactLocation(ArtifactId artifactId)
 
     int responseCode = response.getResponseCode();
     if (responseCode != HttpURLConnection.HTTP_OK) {
-      switch (responseCode) {
-        // throw retryable error if app fabric is not available, might be due to restarting
-        case HttpURLConnection.HTTP_BAD_GATEWAY:
-        case HttpURLConnection.HTTP_UNAVAILABLE:
-        case HttpURLConnection.HTTP_GATEWAY_TIMEOUT:
-          throw new ServiceUnavailableException(
-            Constants.Service.APP_FABRIC_HTTP,
-            Constants.Service.APP_FABRIC_HTTP + " service is not available with status " + responseCode);
-        case HttpURLConnection.HTTP_NOT_FOUND:
-          throw new ArtifactNotFoundException(artifactId);
+      if (io.cdap.cdap.api.common.Constants.HTTP_SERVER_ERROR_CODES.contains(responseCode)) {
```

Comment:
I am wondering if we should rename class to prevent collisions

Reply:
renamed to ConstantsUtil to prevent collision with several other classes named Constants in other packages.